middleware/kubernetes/kubernetes.go

```diff
@@ -200,16 +202,16 @@ func (k *Kubernetes) Records(name string, exact bool) ([]msg.Service, error) {
 	// Abort if the namespace does not contain a wildcard, and namespace is not published per CoreFile
 	// Case where namespace contains a wildcard is handled in Get(...) method.
 	if (!nsWildcard) && (len(k.Namespaces) > 0) && (!dnsstrings.StringInSlice(namespace, k.Namespaces)) {
-		return nil, nil
+		return nil, nsUnexposedErr
 	}
 
 	k8sItems, err := k.Get(namespace, nsWildcard, serviceName, serviceWildcard, typeName)
 	if err != nil {
 		return nil, err
 	}
-	if k8sItems == nil {
+	if len(k8sItems) == 0 {
```

Comment:
k8s will return an empty list if there are no services...  if that's what you mean.